src/lib/components/TipTap.svelte

```diff
@@ -0,0 +1,284 @@
+<script lang="ts">
+	import { Editor } from "@tiptap/core";
+	import StarterKit from "@tiptap/starter-kit";
+	import Icon from "@iconify/svelte";
+	import Underline from "@tiptap/extension-underline";
+
+	let {
+		content,
+		onupdate,
+		editable = true,
+		editor = $bindable<Editor>(),
+		onclick,
+		...args
+	}: {
+		content: string;
+		editable?: boolean;
+		onclick?: (e: MouseEvent) => void;
+		onupdate?: (editor: Editor) => void;
+		editor?: Editor;
+		class?: string;
+	} = $props();
+	let element = $state<Element>();
+
+	let is_bold = $state<boolean>(false);
+	let is_italic = $state<boolean>(false);
+	let is_underline = $state<boolean>(false);
+	let can_undo = $state<boolean>(false);
+	let can_redo = $state<boolean>(false);
+	$effect(() => {
+		if (element) {
+			editor = new Editor({
+				element: element,
+				extensions: [StarterKit, Underline],
+				content: content,
+				editable: editable,
+				onUpdate: () => {
+					onupdate && onupdate(editor);
+				},
+				onTransaction: () => {
+					// force re-render so `editor.isActive` works as expected
+					editor = editor;
+					is_bold = editor.isActive("bold");
+					is_italic = editor.isActive("italic");
+					is_underline = editor.isActive("underline");
+					can_undo = editor.can().undo();
+					can_redo = editor.can().redo();
+				},
+			});
+		}
+		return () => {
+			editor?.destroy();
+		};
+	});
+</script>
+
+<!-- svelte-ignore a11y_click_events_have_key_events -->
+<div
+	bind:this={element}
+	onclick={(e: MouseEvent) => {
+		onclick && onclick(e);
+		editor.chain().focus().run();
+	}}
+	role="button"
+	tabindex="0"
+	class={`
+		${editable && `peer editable`}
+
+		note-content overflow-y-scroll h-full
```

Comment:
make this class list: note-content overflow-y-auto h-200

overflow-y-auto prevents the scrollbar from showing up in chrome when not necessary.

h-200 makes the editor taller by default, easier to read.